tests/Knp/Menu/Tests/Twig/MenuExtensionTest.php

```diff
@@ -130,17 +149,39 @@ private function getHelperMock(array $methods)
         ;
     }
 
+    private function getManipulatorMock(array $methods)
+    {
+        return $this->getMockBuilder('Knp\Menu\Util\MenuManipulator')
+            ->disableOriginalConstructor()
+            ->setMethods($methods)
+            ->getMock()
+        ;
+    }
+
+    private function getMatcherMock()
+    {
+        return $this->getMock('Knp\Menu\Matcher\MatcherInterface');
+    }
+
     /**
      * @param string                $template
      * @param \Knp\Menu\Twig\Helper $helper
      *
      * @return \Twig_Template
      */
-    private function getTemplate($template, $helper)
+    private function getTemplate($template, $helper, $matcher = null, $menuManipulator = null)
     {
+        if (null === $matcher) {
+            $matcher = $this->getMatcherMock();
+        }
+
+        if (null === $menuManipulator) {
+            $menuManipulator = $this->getManipulatorMock(array());
+        }
```

Comment:
I would rather keep pass ``null`` to optional dependencies when they are not provided by tests